src/Acuminator/Acuminator.Analyzers/StaticAnalysis/SuppressCommentFix.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Acuminator.Utilities.Common;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Acuminator.Analyzers.StaticAnalysis
+{
+	[Shared]
+	[ExportCodeFixProvider(LanguageNames.CSharp)]
+	public class SuppressCommentFix : CodeFixProvider
+	{
+		private const string _comment = @"// Acuminator disable once {0} {1} [Justification]";
+		private const string _diagnosticName = @"Suppress diagnostic {0}";
+
+		private static readonly ImmutableArray<string> _fixableDiagnosticIds;
+
+		static SuppressCommentFix()
+		{
+			Type diagnosticsType = typeof(Descriptors);
+			var propertiesInfo = diagnosticsType.GetRuntimeProperties();
+
+			_fixableDiagnosticIds = propertiesInfo
+				.Where(x => x.PropertyType == typeof(DiagnosticDescriptor))
+				.Select(x =>
+				{
+					var descriptor = (DiagnosticDescriptor)x.GetValue(x);
+					return descriptor.Id;
+				})
+				.ToImmutableArray();
+		}
+
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			_fixableDiagnosticIds;
+
+		public override Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			return Task.Run(() =>
```

Comment:
From my point of view the code of the method is more readable if return Task.CompletedTask in the end of the method instead of passing all code in Task.Run

Reply:
Method rewritten.